/*
 * This file was automatically generated by EvoSuite
 * Sat May 16 12:38:58 UTC 2020
 */

package org.mockito.internal.invocation;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;
import org.mockito.internal.invocation.Invocation;
import org.mockito.internal.invocation.MockitoMethod;
import org.mockito.internal.invocation.realmethod.RealMethod;
import org.mockito.internal.reporting.PrintSettings;

@RunWith(EvoRunner.class) @EvoRunnerParameters(useVFS = true, useJEE = true) 
public class Invocation_ESTest extends Invocation_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      MockitoMethod mockitoMethod0 = mock(MockitoMethod.class, new ViolatedAssumptionAnswer());
      doReturn((String) null).when(mockitoMethod0).toString();
      doReturn((Class) null).when(mockitoMethod0).getReturnType();
      doReturn(false).when(mockitoMethod0).isVarArgs();
      PrintSettings printSettings0 = mock(PrintSettings.class, new ViolatedAssumptionAnswer());
      Object object0 = new Object();
      Invocation invocation0 = new Invocation(mockitoMethod0, mockitoMethod0, (Object[]) null, 8, (RealMethod) null);
      Invocation.isToString(invocation0);
      invocation0.getRawArguments();
      try { 
        invocation0.callRealMethod();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.mockito.internal.invocation.Invocation", e);
      }
  }
}
